files/private-chef-cookbooks/private-chef/recipes/add_ons_local.rb

```diff
@@ -17,8 +17,19 @@
 
 addon_path = node['private_chef']['addons']['path']
 node['private_chef']['addons']['packages'].each do |pkg|
-  # find the newest package of each name
-  pkg_file = Dir["#{addon_path}/#{pkg}*.#{package_suffix}"].sort_by{ |f| File.mtime(f) }.last
+  if ::File.directory?(addon_path)
+    # find the newest package of each name if 'addon_path' is a directory
+    pkg_file = Dir["#{addon_path}/#{pkg}*.#{package_suffix}"].sort_by{ |f| File.mtime(f) }.last
+  else
+    if File.basename(addon_path) =~ /#{pkg}/
+      # use the full package path if it contains the package name
+      pkg_file = addon_path
+    else
```

Comment:
Why not just assume we have the right file if someone passes in a full path, perhaps with a ::File.exists? test?  Looking at the parent directory doesn't seem like intuitive behavior to me.